src/fuzz/fuzz-calendarspec.c

```diff
@@ -0,0 +1,28 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "alloc-util.h"
+#include "calendarspec.h"
+#include "fd-util.h"
+#include "fuzz.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        _cleanup_(calendar_spec_freep) CalendarSpec *cspec = NULL;
+        _cleanup_free_ char *str = NULL, *p = NULL;
+
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        str = malloc0(sizeof(*data) * (size + 1));
+        assert_se(str != NULL);
+        memcpy_safe(str, data, size);
+        str[size] = '\0';
```

Comment:
Please use `memdup_suffix0` instead.

Reply:
I suspected there's a function dedicated just for that, will do, thanks!

Reply:
Also, is there any particular reason why `memcpy_safe` (and probably `memcmp*/memset*` as well)  is defined in `basic/util.h` whereas the rest of the mem-related functions is in `basic/alloc-util.h`?

Reply:
A long time ago everything was in util.h, and some things were moved out. It's all a bit of a mess.